src/NuGet.Core/NuGet.Commands/RestoreCommand/SourceRepositoryDependencyProvider.cs

```diff
@@ -539,10 +549,29 @@ private void AddLocalV3ResourceOptions(FindPackageByIdResource resource)
                     logger,
                     cancellationToken);
             }
-            catch (FatalProtocolException e) when (_ignoreFailedSources)
+            catch (FatalProtocolException e)
             {
-                await LogWarningAsync(id, e);
-                return null;
+                if (_ignoreFailedSources)
+                {
+                    await LogWarningAsync(id, e);
+                    return null;
+                }
+                else
+                {
+                    NuGetLogCode logCode = e.LogCode;
+
+                    RestoreLogMessage logMessage = null;
+                    if (e.InnerException != null)
+                    {
+                        logMessage = RestoreLogMessage.CreateError(logCode, e.Message + Environment.NewLine + e.InnerException.Message);
```

Comment:
another place to remove string concatenation

Reply:
yep. next iteration. same feedback as damon, after you found it in the other place first.